scripts/lib/links/ExternalLink.ts

```diff
@@ -0,0 +1,58 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+export class ExternalLink {
+  readonly value: string;
+  readonly originFiles: Set<string>;
+
+  /**
+   * linkString: Link as it appears in source file
+   * originFiles: Paths to source file containing link
+   */
+  constructor(linkString: string, originFiles: string[]) {
+    if (!linkString.startsWith("http")) {
+      throw new Error(
+        `Invalid ExternalLink, should start with 'http': ${linkString}`,
+      );
+    }
+
+    // We strip anchors.
+    this.value = linkString.split("#", 2)[0];
+    this.originFiles = new Set(originFiles);
+  }
+
+  /**
+   * Returns an error message if link failed.
+   */
+  async check(): Promise<string | undefined> {
+    let error: string = "";
+    try {
+      const response = await fetch(this.value, {
+        headers: { "User-Agent": "qiskit-documentation-broken-links-finder" },
+      });
```

Comment:
Not technically part of this PR, but should we be using `method: "HEAD"` here to avoid downloading the body? Might make things faster and reduce rate limiting if we're not doing that already.